tests/unit/ct-completion.test.ts

```diff
@@ -0,0 +1,145 @@
+
+  // disable the stub, which means the real code will return param1 'a' or 97
+  vmock_vmock_examples_simpleFunction_enable_disable(vmock_session, false);
+  VASSERT_EQ (97, simpleFunction ('a', 1.0));
+
+}
+`;
+
+const timeout = 30_000; // 30 seconds
+
+describe("Testing pythonUtilities (valid)", () => {
+  afterEach(() => {
+    vi.restoreAllMocks();
+  });
+
+  test(
+    "validate coded test completion for lines of interest (// vmock) and extra text. ",
+    async () => {
+      // To achieve 100% coverage
+      // --> Need to mock getChoiceDataFromPython, since "extraText" is not implemented yet
+      const pythonUtilities = await import("../../server/pythonUtilities");
+      vi.spyOn(pythonUtilities, "getChoiceDataFromPython").mockReturnValue({
+        choiceKind: "File",
+        choiceList: ["unit", "Prototype-Stubs"],
+        extraText: "some extra data",
+        messages: [],
+      });
+
+      const tstText = unitTst;
+      const lineToComplete = "// vmock";
+
+      const completionPosition = getCompletionPositionForLine(
+        lineToComplete,
+        tstText
+      );
+
+      // Flag that we are dealing with coded tests
+      const cppTestFlag = { cppNode: true, lineSoFar: lineToComplete };
+      const triggerCharacter = lineToComplete.at(-1);
+
+      const codedTestCompletionData = generateCompletionData(
+        tstText,
+        completionPosition,
+        triggerCharacter,
+        cppTestFlag
+      );
+
+      expect(codedTestCompletionData).toEqual([
+        {
+          additionalTextEdits: [
+            {
+              newText: "// some extra data",
+              range: {
+                end: {
+                  character: 0,
+                  line: 0,
+                },
+                start: {
+                  character: 0,
+                  line: 0,
+                },
+              },
+            },
+          ],
+          data: 0,
+          detail: "",
+          kind: 1,
+          label: "unit",
+        },
+        {
+          data: 1,
+          detail: "",
+          kind: 1,
+          label: "Prototype-Stubs",
+        },
+      ]);
+    },
+    timeout
+  );
+
+  test(
+    "validate coded test completion for lines of interest (// vmock) and without extra text. ",
+    async () => {
+      // No mock for this test to let it excercise the actual python but without extra text
+      const tstText = unitTst;
+      const lineToComplete = "// vmock";
+
+      const completionPosition = getCompletionPositionForLine(
+        lineToComplete,
+        tstText
+      );
+
+      // Flag that we are dealing with coded tests
+      const cppTestFlag = { cppNode: true, lineSoFar: lineToComplete };
+      const triggerCharacter = lineToComplete.at(-1);
+
+      const codedTestCompletionData = generateCompletionData(
+        tstText,
+        completionPosition,
+        triggerCharacter,
+        cppTestFlag
+      );
+
+      expect(codedTestCompletionData).toEqual([
```

Comment:
Okay, I think we need to "go deeper" on the completion.

We need to send _three_ completitons:

* `// vmock` (we should check we get a list of units)
* `// vmock unit` (we should check we get `bar`; the only function in `unit`)
* `// vmock unit bar` (we should check we get the mock snippet; see below)

For the last one, running:

```
/home/avj/vcast/vc24__101394_store_mock_info/vpython \
    /home/avj/clones/vector-vscode-vcast/coded_mock/python/testEditorInterface.py \
    choiceList-ct \
    /home/avj/clones/vector-vscode-vcast/coded_mock_tests_denis/tests/unit/vcast/TEST \
    "// vmock unit bar"
```

Gives:

```
{
  "choiceKind": "Snippet",
  "choiceList": [
    "\nint vmock_unit_bar(::vunit::CallCtx<> vunit_ctx, int z) {\n  // Enable Stub: vmock_unit_bar_enable_disable(vmock_session);\n  // Disable Stub: vmock_unit_bar_enable_disable(vmock_session, false);\n\n  // Insert mock logic here!\n}\nvoid vmock_unit_bar_enable_disable(vunit::MockSession &vmock_session, bool enable = true) {\n    using vcast_mock_rtype = int ;\n    vcast_mock_rtype (*vcast_fn_ptr)(int)  = &bar;\n    vmock_session.mock <vcast_mock_rtype (*)(int)> ((vcast_mock_rtype (*)(int))vcast_fn_ptr).assign (enable ? &vmock_unit_bar : nullptr);\n}\n// end of mock for: vmock_unit_bar -------------------------------------------------------------------------------------\n\n"
  ],
  "extraText": "",
  "messages": []
}
```

Reply:
Added test cases for the other completions in 9038de54b41c65eec73f8f620cbae868eb5a3936 .